### PR TITLE
Update slices.SortFunc for Go 1.21

### PR DESCRIPTION
**Pull request by evanj**
**What changed?**

The golang.org/x/exp module was updated to use the same function signature as Go 1.21's upcoming slices.SortFunc function. This changed the return value from a bool (less than) to an int (compare to). Update Temporal to the latest version and fix the uses.

**Why?**

For more details about the upstream change, see the following issue and commit:

- https://github.com/golang/go/issues/61374
- https://cs.opensource.google/go/x/exp/+/302865e7556b4ae5de27248ce625d443ef4ad3ed

**How did you test it?**

I ran the tests locally on my machine.

**Potential risks**

Worst case I may have switched the sort order somewhere accidentally.

**Is hotfix candidate?**

No

### CODE REVIEW

**Comment by evanj** [comment]
Hm I thought Datadog has a CLA already. Let me ask around internally about that.

**Comment by evanj** [comment]
Hmmm Datadog *definitely* has a Temporal CLA signed already. I think the problem may be that I need to open the pull request from the DataDog github account; let me see if I can work out the permissions I may need.

File: service/history/queues/action_pending_task_count.go [hunk 1]
```diff
@@ -126,10 +126,11 @@ func (a *actionQueuePendingTask) gatherStatistics(
 		})
 	}
 	for _, sliceList := range a.slicesPerNamespace {
-		slices.SortFunc(sliceList, func(this, that Slice) bool {
+		slices.SortFunc(sliceList, func(this, that Slice) int {
 			thisMin := this.Scope().Range.InclusiveMin
 			thatMin := that.Scope().Range.InclusiveMin
-			return thisMin.CompareTo(thatMin) > 0
+			// sort in largest to smallest order
+			return thisMin.CompareTo(thatMin) * -1
```
Comment (yycptt) [inline service/history/queues/action_pending_task_count.go:133]:
```suggestion
			return thatMin.CompareTo(thisMin)
```

**Review by yycptt — APPROVED** [review]
cc @yiminc re. the CLA

File: service/history/queues/action_pending_task_count.go (hunk 1 above)
Comment (evanj) [reply service/history/queues/action_pending_task_count.go:133]: great suggestion thanks!

**Comment by evanj** [comment]
Re CLA: Do you want to me to open this pull request again, from the Datadog github account? I believe I can do that. This work was done as a Datadog employee, so the CLA should be signed.

I just updated the branch in the hopes that the test failures were flakes.

**Comment by yiminc** [comment]
@evanj we do need to get the CLA signed in order to get this pr merged.

**Pull request merged**